nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/util/RateLimit.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.processors.slack.util;
+
+import org.apache.nifi.logging.ComponentLog;
+
+import java.util.Date;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class RateLimit {
+    private final AtomicLong nextRequestTime = new AtomicLong(0L);
+    private final ComponentLog logger;
+
+    public RateLimit(final ComponentLog logger) {
+        this.logger = logger;
+    }
+
+    public void retryAfter(final int seconds) {
```

Comment:
Instead of having this as a number of seconds, what about using `java.time.Duration`?

Reply:
Yeah, good call.